### PR TITLE
generate metadata files - 18

### PR DESCRIPTION
**Pull request by gaurabdg**

### CODE REVIEW

File: src/main/resources/UnnecessaryParenthesesCheck.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ are unnecessary based on the types in an expression.  It also doesn't know
+ about operator precedence and associativity; therefore it won't catch
+ something like
+ </p><p>
+ In the above case, given that <em>a</em>, <em>b</em>, and <em>c</em> are
+ all {@code int} variables, the parentheses around {@code a + b}
+ are not needed.
+ </p>]]></description>
+      <properties>
+        <property name="tokens" type="int[]" default-value="EXPR,IDENT,NUM_DOUBLE,NUM_FLOAT,NUM_INT,NUM_LONG,STRING_LITERAL,LITERAL_NULL,LITERAL_FALSE,LITERAL_TRUE,ASSIGN,BAND_ASSIGN,BOR_ASSIGN,BSR_ASSIGN,BXOR_ASSIGN,DIV_ASSIGN,MINUS_ASSIGN,MOD_ASSIGN,PLUS_ASSIGN,SL_ASSIGN,SR_ASSIGN,STAR_ASSIGN,LAMBDA">
```
Comment (romani) [inline src/main/resources/UnnecessaryParenthesesCheck.xml:20]: `type="int[]"`  vs `default-value="EXPR,IDENT,NUM_D...`
in nonsense

please make
`<property name="tokens" type="String[]"` 
see setter.
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java#L95

please put in file accessible tokens and  required tokens if we do not plan to let plugin instantiate Check to get all that details in runtime himself.

Comment (gaurabdg) [reply src/main/resources/UnnecessaryParenthesesCheck.xml:20]: > `type="int[]"` vs `default-value="EXPR,IDENT,NUM_D...`
> in nonsense
> 
> please make
> `<property name="tokens" type="String[]"`
> see setter.
> https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java#L95

Yes, but you instructed to use `int[]` in https://github.com/checkstyle/checkstyle/pull/8416#discussion_r451569007 as tokens are inherently integer in value.

Comment (romani) [reply src/main/resources/UnnecessaryParenthesesCheck.xml:20]: Looks like it was my mistake, if setter use String [], we should use `String[]` as type.

Comment (gaurabdg) [reply src/main/resources/UnnecessaryParenthesesCheck.xml:20]: Sure, no problem, I'll update them.

File: src/main/resources/UnnecessarySemicolonInEnumerationCheck.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<checkstyle-metadata>
+  <module>
+    <check name="UnnecessarySemicolonInEnumerationCheck" fully-qualified-name="com.puppycrawl.tools.checkstyle.checks.coding.UnnecessarySemicolonInEnumerationCheck">
+      <description><![CDATA[]]></description>
+    </check>
+  </module>
```
Comment (strkkk) [inline src/main/resources/UnnecessarySemicolonInEnumerationCheck.xml:8]: why all info here is empty?
It has violation keys at least

File: src/main/resources/UpperEllCheck.xml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<checkstyle-metadata>
+  <module>
+    <check name="UpperEllCheck" fully-qualified-name="com.puppycrawl.tools.checkstyle.checks.UpperEllCheck">
+      <description><![CDATA[]]></description>
+    </check>
+  </module>
```
Comment (strkkk) [inline src/main/resources/UpperEllCheck.xml:8]: same here

**Pull request closed without merging**